src/HTTPBackend.js

```diff
@@ -30,20 +30,46 @@ export default function(options) {
 
 	function parseResponse(request, response) {
 		return new Promise(function(resolve, reject) {
-			response.json().then((json) => {
-
-				let headers = { ...request.header };
+			let headers = { ...request.header };
+			delete headers.resource;
+			delete headers.parameters;
+			delete headers.queryParameters;
+
+			response.text().then(text => {
+				try {
+					const json = JSON.parse(text);
+					resolve({
+						header: { ...headers, statusCode: response.status },
+						body: json
+					});
+				} catch (err) {
+					resolve({
+						header: { ...headers, statusCode: response.status },
```

Comment:
There is an interesting case here where the response _might_ be a 200, yet have invalid json. So rxws would route the text response as a non-error instead.